### PR TITLE
ITN: rotate stake and pre-generate funding keys

### LINKED ISSUE #13904: Rotate stake during experiment
To make sure epoch ledger changes substantially between epoch, need to implement a mechanism to rotate stake between accounts.

Make this part of generator/orchestrator tool.

**Orchestrator interface**

Add a command `rotate-balance` with parameters:

* `key-dir`, directory with `n` private keys 
* `ratio`, ratio of each private key's balance to be used in the rotation
* `mapping`, array of receiver indexes
   * size of array equals `n`
   * each index is from `0` to `n - 1`  inclusive
   * value `j` at index `i` means that in this rotation key `i` sends a payment to key `j`

**Generator interface**

Add support of the following options in the generator:

* `--rotation-key-dir`, directory with private keys (if not provided, no rotations are generated)
* `--rotation-key-num`, number of keys in key dir
   * if the flag isn't provided or provides `0` value, `key dir` is to be examined by generator and number of private keys in that dir is to be used
* `--rotation-permutation`, if `mapping`s generated should be permutations instead of random index lists
* `--rotations-per-round`, number of rotations to perform each round (default: 1)

**Design**

Orchestrator executes the command in the following way:

For each key in the `mapping`:

1. Imports key into the rest server
2. Sends payment according to mapping

In case of error, other rest servers are tried (until total of 5 attempts per payment aren't exceeded).

**Other changes**

Add `RotationRestServers` (typed as string list) to `config.json` to enable payment sending.

### PR DESCRIPTION
**Pull request by georgeee**
Explain your changes:

* Implement stake rotations (to ensure each epoch >33% of stake changes hands)
  * Transactions are randomly generated between BP keys and are executed on each round
  * Design of stake rotations is unusual in that it requires a list of BP public keys and a list of rest servers (which have the corresponding secret keys preloaded) to be specified for experiment generation
  * Design above was chosen in order to make this change isolated to Orchestrator/Generator, without any modification of Mina Daemon code

* (NIT) Add a simple test for keyloader
* Allow funding keys to be generated from initial "seed" secret key
   * This finalizes workflow of funding, making it easy to use (because only one secret key needs to be provided to start the process) and super-efficient (by performing funding from 20 or more keys in parallel) at the same time

Explain how you tested your changes:
* Tested by running experiments with updated generator/orchestrator on the cluster

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? List them

* Closes #13904

### CODE REVIEW

**Comment by georgeee** [comment]
!ci-build-me

File: src/app/itn_orchestrator/src/keyloader.go [hunk 1]
```diff
@@ -66,54 +65,61 @@ type KeyloaderParams struct {
...
+	if err != nil {
+		return nil, errors.Join(err, fmt.Errorf("failed decoding private key from file %s", fname))
 	}
+	return sk, nil
+}
+
+func DecodePrivateKey(bs []byte, password []byte) ([]byte, error) {
+	var box secretBox
+	json.Unmarshal(bs, &box)
+	if box.Box_primitive != "xsalsa20poly1305" || box.Pw_primitive != "argon2i" {
```
Comment (ghost-not-in-the-shell) [inline src/app/itn_orchestrator/src/keyloader.go:88]: What's those?

**Review by ghost-not-in-the-shell — APPROVED** [review]

File: src/app/itn_orchestrator/src/keyloader.go (hunk 1 above)
Comment (georgeee) [reply src/app/itn_orchestrator/src/keyloader.go:88]: I implemented parsing of our password-protected private key format in Go (it wasn't hard and is a more robust solution comparing to relying on standalone `mina` executable).

This code is there for a long time, in this PR I only move it from one larger function to a smaller one (and add a simple unit test).

**Comment by georgeee** [comment]
!ci-build-me

**Comment by nholland94** [comment]
!approved-for-mainnet

**Pull request merged**